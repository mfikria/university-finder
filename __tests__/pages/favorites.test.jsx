/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import Favorites from '../../pages/favorites'

const mockSession = {
  user: {
    image: null,
    name: "John",
    email: "john@email.com",
  },
  expires: 123213139,
}

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(() => ({
    data: mockSession,
    status: 'loading'
  }))
}))

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({
    pathname: '',
    query: {},
    push: jest.fn()
  }))
}))

describe('Favorites', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Favorites />)
    expect(asFragment()).toMatchSnapshot()
  })
})
