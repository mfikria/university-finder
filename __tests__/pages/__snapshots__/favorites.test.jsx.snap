// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
  >
    <header
      class="ant-layout-header siteHeader"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col"
          style="flex-basis: auto;"
        >
          <ul
            class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-overflow-item ant-menu-item"
              data-menu-id="rc-menu-uuid-test-/"
              role="menuitem"
              style="opacity: 1; order: 0;"
              tabindex="-1"
            >
              <svg
                class="ant-menu-item-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
              <span
                class="ant-menu-title-content"
              >
                Home
              </span>
            </li>
            <li
              class="ant-menu-overflow-item ant-menu-item"
              role="menuitem"
              style="opacity: 1; order: 1;"
              tabindex="-1"
            >
              <svg
                class="ant-menu-item-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M880 112c-3.8 0-7.7.7-11.6 2.3L292 345.9H128c-8.8 0-16 7.4-16 16.6v299c0 9.2 7.2 16.6 16 16.6h101.7c-3.7 11.6-5.7 23.9-5.7 36.4 0 65.9 53.8 119.5 120 119.5 55.4 0 102.1-37.6 115.9-88.4l408.6 164.2c3.9 1.5 7.8 2.3 11.6 2.3 16.9 0 32-14.2 32-33.2V145.2C912 126.2 897 112 880 112zM344 762.3c-26.5 0-48-21.4-48-47.8 0-11.2 3.9-21.9 11-30.4l84.9 34.1c-2 24.6-22.7 44.1-47.9 44.1zm496 58.4L318.8 611.3l-12.9-5.2H184V417.9h121.9l12.9-5.2L840 203.3v617.4z"
                />
              </svg>
              <span
                class="ant-menu-title-content"
              >
                Subscription
              </span>
            </li>
            <li
              class="ant-menu-overflow-item ant-menu-item"
              role="menuitem"
              style="opacity: 1; order: 2;"
              tabindex="-1"
            >
              <svg
                class="ant-menu-item-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <span
                class="ant-menu-title-content"
              >
                Favorites
              </span>
            </li>
            <li
              aria-hidden="true"
              class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
              role="none"
              style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
            >
              <div
                aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-menu-submenu-title"
                data-menu-id="rc-menu-uuid-test-rc-menu-more"
                role="menuitem"
                tabindex="-1"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
                <i
                  class="ant-menu-submenu-arrow"
                />
              </div>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display: none;"
          />
        </div>
        <div
          class="ant-col accountMenu"
        >
          <span
            class="ant-dropdown-trigger"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span
              class="email"
            >
              john@email.com
            </span>
          </span>
        </div>
      </div>
    </header>
    <main
      class="ant-layout-content siteLayout"
    >
      <div
        class="siteLayoutBackground"
      >
        <h3
          class="ant-typography"
        >
          Your Favorite Universities
        </h3>
        <div
          class="scrollableList"
          id="scrollableList"
        >
          <div
            class="ant-list ant-list-split ant-list-loading"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div>
                <div
                  class="ant-spin ant-spin-spinning"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
              <div
                class="ant-spin-container ant-spin-blur"
              >
                <div
                  style="min-height: 53px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </section>
</DocumentFragment>
`;
